paramiko/file.py

```diff
@@ -372,7 +372,10 @@ def write(self, data):
 
         :param str/bytes data: data to write
         """
-        data = b(data)
+        if isinstance(data, text_type):
+            # GZ 2017-05-25: Accepting text on a binary stream unconditionally
+            # cooercing to utf-8 seems questionable, but compatibility reasons?
```

Comment:
... I also typoed my first comment, was "`str` for py2, `unicode` for py2", first py2 should have been py3, I just corrected it ... that couldn't have helped ...